pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/JavaInstanceRunnable.java

```diff
@@ -324,7 +327,11 @@ private void processResult(Record srcRecord,
                 srcRecord.fail();
             } else {
                 if (result1.getResult() != null) {
-                    sendOutputMessage(srcRecord, result1.getResult());
+                    try {
+						sendOutputMessage(srcRecord, result1.getResult());
+					} catch (Exception ex) {
+						throw new CompletionException(ex);
```

Comment:
Solution based roughly on [this](https://stackoverflow.com/questions/44409962/throwing-exception-from-completablefuture) suggestion.